src/datasets/streaming.py

```diff
@@ -112,3 +111,12 @@ def extend_dataset_builder_for_streaming(builder: "DatasetBuilder"):
                 internal_import_name = imports[1]
                 internal_module_name = ".".join(builder.__module__.split(".")[:-1] + [internal_import_name])
                 extend_module_for_streaming(internal_module_name, use_auth_token=builder.use_auth_token)
+
+    # we need to patch autofolder.py manually because it's a parent of ImageFolder and AudioFolder,`
+    # and it contains generation of examples for them (which uses `os` functions)
+    # but it's not patched automatically because it's not instantiated
+    from .packaged_modules.base.autofolder import AutoFolder
+
+    autofolder_module = [cls.__module__ for cls in type(builder).__mro__ if cls is AutoFolder]
+    if autofolder_module:
+        extend_module_for_streaming(autofolder_module[0], use_auth_token=builder.use_auth_token)
```

Comment:
IMO it would be cleaner to remove this and allow patching internal imports in the packaged modules as long as these imports don't refer to something that's outside the `packaged_modules` subpackage.

This can be done by removing [this](https://github.com/huggingface/datasets/blob/18287ed524ac79f538abf77183f906e91d0de3de/src/datasets/streaming.py#L109) check and adding a test for internal imports to [this](https://github.com/huggingface/datasets/blob/main/tests/test_dataset_scripts.py) file

Reply:
that's what I tried to do in the first place but it appeared that [`get_import`](https://github.com/huggingface/datasets/blob/main/src/datasets/utils/py_utils.py#L514) do not work with imports starting with `..` and @lhoestq and I decided that it would be easier to hardcode it with a comment instead of modifying this function and everything that uses it. 

I tried to modify `get_import` in a non-breaking way here [bc1fb3d](https://github.com/huggingface/datasets/pull/4530/commits/bc1fb3d6937b194011a28b15df272801cc10187c), but I don't like it that way. Also, it breaks tests on windows CI and I can't figure out why. so maybe we can leave it hardcoded? or do you have any better ideas?

Reply:
ah tests shouldn't work this way either because in tests it would be imports from datasets library, not internal. so I would suggest to revert to hardcoded patching. cc @lhoestq

Reply:
@mariosasko are you ok with this solution? https://github.com/huggingface/datasets/pull/4530/commits/bfecab46c0aa2adea8adb014204ad8f26fd4c4bc

Reply:
I guess it's ok to hardcode this logic considering relative imports outside a builder package are only possible here, so a more generic approach is not needed at the moment.